comp/rdnsquerier/impl/rdnsquerier.go

```diff
@@ -48,16 +78,83 @@ func NewComponent(reqs Requires) (Provides, error) {
 	}, nil
 }
 
+// GetHostname attempts to use reverse DNS lookup to resolve the hostname for the given IP address.
+// If the IP address is in the private address space and the lookup is successful then the updateHostname
+// function will be called with the hostname.
+func (q *rdnsQuerierImpl) GetHostname(ipAddr []byte, updateHostname func(string)) {
```

Comment:
It doesn't look like this function blocks, so doesn't this create a race condition? I.e. if someone wrote
```go
foo := Something()
q.GetHostname(someAddr, func(s string) { foo.hostname = s })
fmt.Printf("Hostname is %q", foo.hostname)
```
it would be random whether it printed before or after the hostname was set?

Reply:
Yes, that is correct, it is asynchronous.  The use case for NetFlow is that flows are aggregated for some period of time before being flushed to the epforwarder.  We don't want to block or slow down the netflow aggregation/flushing, so we make a best attempt to resolve the IP address between the time that a flow is first added to the aggregator and when it is flushed.  There should almost always be enough time to do so, but if not we will have netflow event(s) that do not have reverse dns enrichment.  This will also be improved somewhat when the caching layer is added, such that if the ip address resolution has been cached it will synchronously be added to the aggregated flow when it is initially added to the aggregator.  I changed the method name to GetHostnameAsync() and updated the comment a bit.